### PR TITLE
Add gentle failover test

### PR DESCRIPTION
**Pull request by ericgribkoff**
Gentle failover is a load balancer feature that will route some traffic to the secondary locality when the number of unhealthy backends exceeds a certain threshold (>50%). This feature requires that the backend VMs remain alive but the gRPC servers report an unhealthy status, which means this PR depends on https://github.com/grpc/grpc-java/pull/7127 for the ability to dynamically set the server to a healthy/unhealthy status.

This PR supersedes https://github.com/grpc/grpc/pull/23143. Currently our only test environment that supports gentle failover is prone to connecting to a TD instance in a different region, requiring that primary/secondary expectations be swapped.

### CODE REVIEW

File: tools/run_tests/run_xds_tests.py [hunk 1]
```diff
@@ -478,61 +531,93 @@ def test_round_robin(gcp, backend_service, instance_group):
...
+                                                     _WAIT_FOR_BACKEND_SEC)
+        finally:
+            set_serving_status(primary_instance_names,
+                               gcp.service_port,
+                               serving=True)
+    except RpcDistributionError as e:
+        if not swapped_primary_and_secondary and is_primary_instance_group(
+                gcp, secondary_instance_group):
+            # Swap expectation of primary and secondary instance groups.
+            test_secondary_locality_gets_requests_on_primary_failure(
```
Comment (gnossen) [inline tools/run_tests/run_xds_tests.py:613]: Just want to make sure I'm understanding this correctly. In your PR description, you say

> Currently our only test environment that supports gentle failover is prone to connecting to a TD instance in a different region, requiring that primary/secondary expectations be swapped.

Is this where you acommodate that possibility?

File: tools/run_tests/run_xds_tests.py [hunk 2]
```diff
@@ -646,7 +751,7 @@ def get_startup_script(path_to_server_binary, service_port):
 sudo apt install -y git default-jdk
 mkdir java_server
 pushd java_server
-git clone https://github.com/grpc/grpc-java.git
+git clone -b xds_server_health https://github.com/ericgribkoff/grpc-java.git
```
Comment (gnossen) [inline tools/run_tests/run_xds_tests.py:754]: Nit: `--single-branch`?

File: tools/run_tests/run_xds_tests.py (hunk 1 above)
Comment (ericgribkoff) [reply tools/run_tests/run_xds_tests.py:619]: Yes! Thanks for the review(s) :-)

File: tools/run_tests/run_xds_tests.py (hunk 2 above)
Comment (ericgribkoff) [reply tools/run_tests/run_xds_tests.py]: Oops, this was just to run tests before my PR had been merged on grpc-java (which is now done).

**Comment by ericgribkoff** [comment]
Successful kokoro run:  https://source.cloud.google.com/results/invocations/79802bd3-c2b2-4c4c-b78e-7d862333ebb3/targets

**Comment by ericgribkoff** [comment]
Kokoro run including gentle failover: https://source.cloud.google.com/results/invocations/0a12337a-3735-4d10-9c13-7137dc4fed4f/targets

**Comment by ericgribkoff** [comment]
https://github.com/grpc/grpc/issues/23239

**Pull request merged**